internal/app/machined/pkg/runtime/v1alpha1/platform/vmware/metadata.go

```diff
@@ -0,0 +1,202 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package nocloud provides the NoCloud platform implementation.
+package vmware
+
+import (
+	"net/netip"
+	"strings"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime"
+	"github.com/siderolabs/talos/pkg/machinery/nethelpers"
+	"github.com/siderolabs/talos/pkg/machinery/resources/network"
+)
+
+const ()
+
+type NetworkConfig struct {
+	InstanceId    string `yaml:"instance-id"`						// TODO: Not used in this implementation
+	LocalHostname string `yaml:"local-hostname"`				// TODO: Not used in this implementation
+	WaitOnNetwork struct {														// TODO: Not used in this implementation
```

Comment:
Talos doesn't block on network, it will reconfigure itself as network information becomes available